src/main/java/org/traccar/protocol/LaipacProtocolDecoder.java

```diff
@@ -209,6 +226,12 @@ protected Object decodeAvrmc(
             return null;
         }
 
+        String deviceModel = null;
+        Device device = getCacheManager().getObject(Device.class, deviceSession.getDeviceId());
+        if (device != null) {
+            deviceModel = device.getModel();
+        }
```

Comment:
```suggestion
        String model = getCacheManager().getObject(Device.class, deviceId).getModel();
```

Reply:
Is it possible that "device" is not still in cache? (Race condition?) I ask this because I'm having (unrelated?) errors in log (I've not debugged them yet) that reads: "WARN: Device xxx cache missing"

Reply:
Anyway, I can follow your suggestion and deffer solving this.

Reply:
There might be some very corner case condition, but in that case we want it to fail.

Reply:
Ok. Let me debug this and I'll open an issue or pull request. Thanks...